buildfiles/cmake/ConfigureCompiler.cmake

```diff
@@ -33,7 +33,9 @@ else()
 	add_compile_options(-fno-exceptions)
 	add_compile_options(-fstack-protector)
 
-	if (COMPILER_ARM)
+	if(USE_NATIVE_INSTRUCTIONS AND COMPILER_SUPPORTS_MARCH_NATIVE)
+		add_compile_options(-march=native)
```

Comment:
This rearrangement will now skip the x86 path. Is this intended?

Reply:
If the `-msse -msee2 -mcx16` are to be taken to mean "simplified x86-64-v2" then yes, `native` should be used instead if it was asked for and is supported.

Reply:
USE_NATIVE_INSTRUCTIONS trumps the -mxxxx identifiers by just specifying -march=native. All those levels get automatically picked up at native level.
x86 toolchains are pretty mature so we only need to be careful about arm, where even now using native instructions breaks things.

Reply:
Maybe this is a good occasion to think about either defaulting to x86-64-v4 for amd64 or adding a string macro like COMPILER_MARCH or something and let people use that to set whatever they want to require from their users hardware.